src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -2126,8 +2128,11 @@ public StructureManager getStructureManager()
 	@Override
 	public @Nullable <T extends Keyed> Registry<T> getRegistry(@NotNull Class<T> tClass)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if (!registry.containsKey(tClass))
+		{
+			registry.put(tClass, RegistryMock.createRegistry(tClass));
+		}
+		return (Registry<T>) registry.get(tClass);
```

Comment:
```suggestion
		return (Registry<T>) registry.computeIfAbsent(tClass, clazz -> RegistryMock.createRegistry(clazz));

Reply:
Not possible to change. Once this method is called due to how Registry class is made all static fields are in fact initialized at once. result would end with concurrent mod ex.